contrib/guix/guix-build

```diff
@@ -73,8 +73,19 @@ mkdir -p "$VERSION_BASE"
 # Build directories should not exist
 ################
 
-# Default to building for all supported HOSTs (overridable by environment)
-export HOSTS="${HOSTS:-x86_64-linux-gnu arm-linux-gnueabihf aarch64-linux-gnu riscv64-linux-gnu powerpc64-linux-gnu powerpc64le-linux-gnu
+# Supported HOSTs:
+# aarch64-linux-gnu
+# arm-linux-gnueabihf
+# arm64-apple-darwin
+# powerpc64-linux-gnu
+# powerpc64le-linux-gnu
+# riscv64-linux-gnu
+# x86_64-apple-darwin
+# x86_64-linux-gnu
+# x86_64-w64-mingw32
+
+# Default to building for some supported HOSTs only (overridable by environment)
+export HOSTS="${HOSTS:-x86_64-linux-gnu aarch64-linux-gnu riscv64-linux-gnu
```

Comment:
Consider b58a3b6e3f78e9224eefab717bf0c6eb84374918

Reply:
you mean in addition to changes in `contrib/guix/guix-build` or as a replacement?

Reply:
Replacement, I think we should keep upstream code intact as much as we can and add restrictions through CI or our containers

Reply:
I disagree, mostly, because very little if any of my automations use the guis-start. They all use guix-buidl

Reply:
`./contrib/containers/guix/scripts/guix-start $(pwd)` should be a drop-in

Reply:
Needs a couple of additional fixes but, yeah, it could work. Pls check #6390.